### ISSUE TITLE
Flaky Conformance Test in K8s v1.21: should complete a service status lifecycle

### ISSUE THREAD

**Issue by mattysweeps** [kubernetes/kubernetes#105028]
#### Which test(s) are flaking: `should complete a service status lifecycle`
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/network/service.go#L2517

#### Testgrid link:
Seeing flakes on this downstream on release 1.21

#### Reason for failure:
Object modifications to the service cause the patch to fail.
```
failed to patch service. Operation cannot be fulfilled on services "test-service-2wkhl": the object has been modified; please apply your changes to the latest version and try again
```

**Comment by liggitt** [comment]
Patch calls are _supposed_ to be retried internally by the API server if the object was modified concurrently... they aren't expected to return conflict errors, so the e2e test seems like it would be correct.

_However_, the REST implementation for Service specifically overrode normal Update behavior and looks like it would prevent that from working properly prior to https://github.com/kubernetes/kubernetes/commit/ccf337657075cd93619c25071f38840ce1acd3b8#diff-675c06306707c1030377e08e295b5ba336db8cf473b655001a2a806a6fd069c5R277 which @thockin just landed in master.

That means that service spec patch requests in the 1.21 and 1.22 conformance tests _have_ to tolerate conflict errors.

**Comment by thockin** [comment]
So has this been flaking forever or did something change?

**Comment by thockin** [comment]
Where is it flaking?  I am not seeing it

**Comment by dcbw** [comment]
@thockin since those PRs for 1.21 and 1.22 are merged, should we close this as fixed?

**Comment by mattysweeps** [comment]
@thockin is this fixed in 1.23+? Did the patch behavior for services change? I don't see any pull requests against the master branch, only v1.21 and v1.22.

Looking at https://github.com/kubernetes/kubernetes/blob/master/test/e2e/network/service.go#L3476
The patch call after `ginkgo.By("patching the service")` is not wrapped in `retry.RetryOnConflict`

**Comment by aojea** [comment]
see comment https://github.com/kubernetes/kubernetes/issues/105028#issuecomment-919550813 
The test does not need to be modified in 1.23 since the logic in the apiserver was fixed by https://github.com/kubernetes/kubernetes/pull/96684
